Update wx to 3.1.0

suraia opened the pull request:
This PR updates wx to version 3.1.0, which is required to compile with GCC 6. It also adds the newest versions for the current (3.0) and previous (2.8) stable branches.

I tried building it in parallel and it seemed to work fine (at least for 3.1.0).

adamjstewart commented on var/spack/repos/builtin/packages/wx/package.py:
```diff
@@ -36,16 +36,20 @@ class Wx(Package):
     version('3.0.1', 'dad1f1cd9d4c370cbc22700dc492da31',
             url="https://sourceforge.net/projects/wxwindows/files/3.0.1/wxWidgets-3.0.1.tar.bz2")
+    version('2.8.12', '2fa39da14bc06ea86fe902579fedc5b1',
+            url="https://sourceforge.net/projects/wxwindows/files/2.8.12/wxWidgets-2.8.12.tar.gz")
```
The best way to handle these changing URLs is to write a `url_for_version` function like in HDF5, but you don't have to.

adamjstewart commented on var/spack/repos/builtin/packages/wx/package.py:
```diff
@@ -36,16 +36,20 @@ class Wx(Package):
                   "--disable-precomp-headers")
 
-        make(parallel=False)
+        make()
```
Looks like @lucjaulmes added `parallel=False` when he first added the package in https://github.com/LLNL/spack/commit/853784d3829166f0d5b797dffcc2a8b63723e350. I'm always cautious with these things, because parallel builds can work for some users and not for others. I would err on the side of caution and keep it. Especially if it is still necessary for the older builds. It just means the build might take a little longer.

suraia commented:
I have updated the versions to always use the GitHub releases and dropped the 2.8.12 version because it does not have any published checksums available (and is from 2011). As far as I could see, nothing depends on the old version.

I have also added a workaround to build newer releases in parallel but keep the serial builds for older versions.
